internal/status/checksum.go

```diff
@@ -32,6 +33,16 @@ func (c *Checksum) IsUpToDate() (bool, error) {
 		return false, err
 	}
 
+	generates, err := glob(c.Dir, c.Generates)
+	if err != nil || len(generates) == 0 {
```

Comment:
Even if the execution is equal, I think that for code clarity this should be two `if` checks:

```go
if err != nil {
	return false, err
}
if len(generates) == 0 {
	return false, nil
}
```

These checks can also be moved as the very first thing to run in this function.

Reply:
Additionally, since `glob` also returns directories, the second check would not work as is if only directories are returned, but no files.

I'm asking myself if we should only return files (and not directories) on `glob` [here](https://github.com/go-task/task/blob/4c295b564a7da0281b70652df09d429eab77684c/internal/status/glob.go#L25). If we do that, we could also removed [these lines](https://github.com/go-task/task/blob/4c295b564a7da0281b70652df09d429eab77684c/internal/status/checksum.go#L61-L63).

Reply:
Thank you for the reviews. Addressed them in the new commits.